lib/debug/config.rb

```diff
@@ -496,7 +498,14 @@ def self.unix_domain_socket_dir
 
   def self.create_unix_domain_socket_name_prefix(base_dir = unix_domain_socket_dir)
     user = ENV['USER'] || 'UnknownUser'
-    File.join(base_dir, "ruby-debug-#{user}")
+    filename = "ruby-debug-#{user}"
+    if !CONFIG[:sock_prefix].nil?
+      filename = "#{CONFIG[:sock_prefix]}-#{filename}"
+    end
+    if !CONFIG[:sock_suffix].nil?
+      filename += "-#{CONFIG[:sock_suffix]}"
+    end
```

Comment:
> On this PR, you only need to rename `PREFIX/SUFFIX` to `SESSION_NAME` (and maybe prefix will be nice).

I think what Koichi meant here was:

```suggestion
    if !CONFIG[:session_name].nil?
      filename = "#{CONFIG[:session_name]}-#{filename}"
    end
```